sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -112,18 +112,6 @@ AggregatorFactory createAggregatorFactory(
             return new DoubleFirstAggregatorFactory(name, fieldName, timeColumn);
           case STRING:
           case COMPLEX:
-            if (type.getComplexTypeName() != null) {
-              switch (type.getComplexTypeName()) {
-                case SerializablePairLongLongComplexMetricSerde.TYPE_NAME:
-                  return new LongFirstAggregatorFactory(name, fieldName, timeColumn);
-                case SerializablePairLongFloatComplexMetricSerde.TYPE_NAME:
-                  return new FloatFirstAggregatorFactory(name, fieldName, timeColumn);
-                case SerializablePairLongDoubleComplexMetricSerde.TYPE_NAME:
-                  return new DoubleFirstAggregatorFactory(name, fieldName, timeColumn);
-                default:
-                  return new StringFirstAggregatorFactory(name, fieldName, timeColumn, maxStringBytes);
-              }
-            }
```

Comment:
Why did this get removed?

Reply:
this was a no-op post return type change that was suggested. Since we were returning the scalar type, it would eventually resolve to the right last/first AggregatorFactory